invenio/modules/indexer/testsuite/test_indexer_engine_washer.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Unit tests for the indexing engine washer."""
+
+from invenio.base.wrappers import lazy_import
+from invenio.testsuite import InvenioTestCase, make_test_suite, run_test_suite
+
+engine_washer = lazy_import('invenio.legacy.bibindex.engine_washer')
+
+
+class TestWashers(InvenioTestCase):
+
+    """Test for washer operations."""
+
+    def test_lower_term_washer(self):
+        """Bibindex engine washer - lower term."""
+        self.assertEqual(
+            engine_washer.lower_index_term("test"),
+            "test"
+        )
+        self.assertEqual(
+            engine_washer.lower_index_term("Test"),
+            "test"
+        )
+
+    def test_lower_term_washer_encoding(self):
+        """Bibindex engine washer - lower term encoding."""
+        self.assertEqual(
+            engine_washer.lower_index_term(u"test"),
+            u"test"
+        )
+        self.assertEqual(
+            engine_washer.lower_index_term("Über"),
+            "über"
```

Comment:
Right. So that would change the behavior of the original function a bit as it was always returning the bytestring. (The test above on line 45 is actually wrong(!), as it should not be `u"test"`, but simply `"test"`.).

The question is, should we allow the unicode string to survive through this function. The obvious answer is yes, but I wonder what it will break further down the line. I could give it a go and see.